sct_ssh.py

```diff
@@ -452,3 +465,94 @@ def gcp_allow_public(user, test_id):
                      zone=i.zone.split('/')[-1])
         click.echo(click.style(f"set netwrok tag 'sct-allow-public' to {get_name(i)}", fg='green'))
+
+
+@click.command("update-scylla-packages", help="Update ScyllaDB cluster packages ...")
+@click.option("--test-id", type=str, help='Find cluster by test-id', required=True)
+@click.option(
+    "-p", "--packages-location", type=str,
+    help='Location where new packages are placed. Can be local directory, s3:// or gs:// urls')
+@click.option('--backend', type=str, help='Backend to search nodes in. Defaults to aws', default="aws")
+@click.option(
+    '-r', '--region', type=str,
+    help="Cloud region to search nodes in. Defaults to eu-west-1", default="eu-west-1")
+def update_scylla_packages(test_id: str, backend: str, region: str, packages_location: str) -> None:
+    """Update scylla DB packages on selected nodes."""
+    if not (packages_location := packages_location or os.getenv('SCT_UPDATE_DB_PACKAGES')):
+        LOGGER.error("No location is provided where new packages are placed")
+        sys.exit(1)
+    if packages_location.startswith('s3') or packages_location.startswith('gs'):
+        packages_location = download_dir_from_cloud(packages_location)
+    if not packages_location.endswith('/'):
+        packages_location += '/'
+
+    instances = select_instance_group(region=region, backends=[backend], test_id=test_id, node_type='scylla-db')
+    if not instances:
+        LOGGER.error("No scylla DB nodes were selected. Exiting")
+        sys.exit(1)
+    for instance in instances:
+        tags = get_tags(instance)
+        instance.update({
+            'Name': tags['Name'],
+            'User': tags['RunByUser']
+        })
+
+    RemoteCmdRunnerBase.set_default_ssh_transport('libssh2')
+    nodes = []
+    for instance in instances:
+        remoter = RemoteCmdRunnerBase.create_remoter(
+            hostname=instance['PublicIpAddress'],
+            user=guess_username(instance),
+            key_file=f"~/.ssh/{instance['KeyName']}")
+        nodes.append(remoter)
+
+    def run_func_parallel(func, node_list):
+        with ThreadPoolExecutor() as executor:
+            futures = [executor.submit(func, node) for node in node_list]
+            results = [future.result() for future in as_completed(futures)]
+
+        return results
+
+    def is_scylla_active(node: RemoteCmdRunnerBase):
+        return node.run('systemctl is-active scylla-server.service', ignore_status=True).stdout.strip() == 'active'
+
+    def stop_scylla(node: RemoteCmdRunnerBase):
+        node.sudo('systemctl stop scylla-server.service')
+        wait.wait_for(func=lambda: not is_scylla_active(node), step=10, timeout=180)
+
+    def start_scylla(node: RemoteCmdRunnerBase):
+        node.sudo('systemctl start scylla-server.service')
+        wait.wait_for(func=lambda: is_scylla_active(node), step=10, timeout=180)
+
+    def update_packages(node: RemoteCmdRunnerBase):
```

Comment:
@fruch
i tried to derive a small dedicated class from Cluster and ScyllaCLuster base classes. It does help to get access and run the update_db_packages related API from those classes.
But the end result is still not OK - there are places where update_db underlying functionality (scylla start/stop on node APIs, waiters, etc.) is coupled with tester object, with events processing, which are not initialized with this hydra command. 

I don't know which is better - leave this duplicated couple bits of update_db_packages functionality, as it is now, or try to further subclass Cluster functionality.

Reply:
you can see `scan-skip-issues` command which enable events for a similar reason.

and I think we are o.k. with doing start/stop scylla as you do here, also in the original implementation, or disable the wait_for_up wait_for_down, there argument for that...

to the very minimum we can extract just the sequence of command that does the updates (without the starting and stop parts)